CHANGELOG.md

```diff
@@ -6,8 +6,14 @@
 
 ## Breaking Changes
 
+- [#624](https://github.com/oauth2-proxy/oauth2-proxy/pull/624) Authentication headers will be stripped in whitelisted requests
+  - This is controlled with the `--skip-auth-strip-headers` flag (default `true`). Can be set to `false` to preserve legacy behavior.
+  - `--pass-basic-auth` will now only control setting the `Authorization` header with basic auth. It no longer controls the `X-Forwarded-*` user headers (already configured with `--pass-user-headers`)
+  - `--pass-basic-auth` now is `false` by default to match related settings defaults.
+
```

Comment:
I wonder if we should merge the logic now, then flip the defaults at a later date. My thinking being that this would be a smoother migration for users and we could do a patch release or two while we build up a few breaking changes to go in, WDYT?

Reply:
Good call.  I flipped the defaults to make this MOSTLY non-breaking.

I left in the change removing `--pass-basic-auth` control of user headers (controlled via `--pass-user-headers` already and already `true` by default as well).

Reply:
I'd rather we made this completely non breaking and then open a PR to contain all the breaking changes and rebase that when we decide we are ready for breaking changes, WDYT?